sale_triple_discount/models/sale_order_line.py

```diff
@@ -0,0 +1,145 @@
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    discount2 = fields.Float(
+        string="Disc. 2 (%)",
+        digits="Discount",
+        default=0.0,
+    )
+    discount3 = fields.Float(
+        string="Disc. 3 (%)",
+        digits="Discount",
+        default=0.0,
+    )
+    discounting_type = fields.Selection(
+        selection=[("additive", "Additive"), ("multiplicative", "Multiplicative")],
+        default="multiplicative",
+        required=True,
+        help="Specifies whether discounts should be additive "
+        "or multiplicative.\nAdditive discounts are summed first and "
+        "then applied.\nMultiplicative discounts are applied sequentially.\n"
+        "Multiplicative discounts are default",
+    )
+
+    def _get_final_discount(self):
+        self.ensure_one()
+        if self.discounting_type == "additive":
+            return self._additive_discount()
+        elif self.discounting_type == "multiplicative":
+            return self._multiplicative_discount()
+        else:
+            raise ValidationError(
+                _("Sale order line %(name)s has unknown discounting type %(dic_type)s")
+                % {"name": self.name, "disc_type": self.discounting_type}
+            )
+
+    def _additive_discount(self):
+        self.ensure_one()
+        discount = sum(self[x] or 0.0 for x in self._discount_fields())
+        if discount <= 0:
+            return 0
+        elif discount >= 100:
+            return 100
+        return discount
+
+    def _multiplicative_discount(self):
+        self.ensure_one()
+        discounts = [1 - (self[x] or 0.0) / 100 for x in self._discount_fields()]
+        final_discount = 1
+        for discount in discounts:
+            final_discount *= discount
+        return 100 - final_discount * 100
+
+    def _discount_fields(self):
+        return ["discount", "discount2", "discount3"]
+
+    @api.depends("discount2", "discount3", "discounting_type")
+    def _compute_amount(self):
+        prev_values = self.triple_discount_preprocess()
+        res = super()._compute_amount()
+        self.triple_discount_postprocess(prev_values)
+        return res
+
+    _sql_constraints = [
+        (
+            "discount2_limit",
+            "CHECK (discount2 <= 100.0)",
+            "Discount 2 must be lower than 100%.",
+        ),
+        (
+            "discount3_limit",
+            "CHECK (discount3 <= 100.0)",
+            "Discount 3 must be lower than 100%.",
+        ),
+    ]
+
+    def _get_triple_discount(self):
+        """Get the discount that is equivalent to the subsequent application
+        of discount, discount2 and discount3"""
+        discount_factor = 1.0
+        for discount in (self.discount, self.discount2, self.discount3):
+            discount_factor *= (100.0 - discount) / 100.0
+        return 100.0 - (discount_factor * 100.0)
```

Comment:
I guess this can be removed since it is not used and it has the same logic as the _multiplicative_discount function.

_get_final_discount has already the proper logic and it also considers the discount type